src/main/java/org/jenkinsci/plugins/ghprb/extensions/comments/GhprbCommentFile.java

```diff
@@ -42,9 +51,50 @@ public String postBuildComment(Run<?, ?> build, TaskListener listener) {
 
             try {
                 String content = null;
-                if (build instanceof Build<?, ?>) {
-                    final FilePath workspace = ((Build<?, ?>) build).getWorkspace();
-                    final FilePath path = workspace.child(scriptFilePathResolved);
+                FilePath workspace;
+                FilePath path;
+
+                // On custom pipelines, build will be an instance of WorkflowRun
+                if (build instanceof WorkflowRun) {
+                    FlowExecution exec = ((WorkflowRun) build).getExecution();
+                    if (exec == null) {
+                        listener.getLogger().println("build was instanceof WorkflowRun but executor was null");
+                    } else {
+                        // We walk the execution flow as a run can have multiple workspaces
+                        FlowGraphWalker w = new FlowGraphWalker(exec);
+                        for (FlowNode n : w) {
+                            if (n instanceof BlockStartNode) {
+                                WorkspaceAction action = n.getAction(WorkspaceAction.class);
+                                if (action != null) {
+                                    String node = action.getNode().toString();
+                                    String nodepath = action.getPath().toString();
+                                    listener.getLogger().println("Remote path is " + node + ":" + nodepath + "\n");
+
+                                    if (action.getWorkspace() == null) {
```

Comment:
should this be assigned to workspace?  or should workspace not be part of this if block?

Reply:
Yep, I've moved the variable definitions around to be more sensible.